reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java

```diff
@@ -2489,4 +2491,88 @@ public void testLoopAndResolver() {
 		loop.disposeLater()
 		    .block(Duration.ofSeconds(30));
 	}
+
+	@Test
+	public void testCustomMetricsWithUriMapper() {
+		disposableServer =
+				HttpServer.create()
+							.port(0)
+							.handle((req, resp) -> resp.sendString(Mono.just("OK")))
+							.wiretap(true)
+							.bindNow();
+
+		final List<String> collectedUris = Collections.synchronizedList(new ArrayList<>());
```

Comment:
Instead of this can we use `Sinks`
`Sinks.Many<String> collectedUris = Sinks.many().replay().all();`